### DISCUSSION TITLE
Full support for Shelly plus 1/1PM add-on

### DISCUSSION THREAD

**Discussion by Kolia56** [arendst/Tasmota#20580] [Ideas]
### PROBLEM DESCRIPTION
It seems that DS18x20 are not yet recognized by this new flavour of Shelly plus 1 with addon. Many different information sources have been crawled with no success so far including this [discussion](https://github.com/arendst/Tasmota/discussions/18885) or hardware discord channel.
Are there any chance to get this device supported? Or is there any specific settings that we should be aware of?
Thank you

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Shelly plus 1 with addon
- [x] Tasmota binary firmware version number used: 13.3.0
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: VSCode and PIO platform
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
15:10:53.311 MQT: stat/tasmota_0B0778/RESULT = {"NAME":"Shelly Plus 1","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1312,1,1,1,1,1,32,224,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":1}
15:10:53.542 MQT: stat/tasmota_0B0778/RESULT = {"Module":{"0":"Shelly Plus 1"}}
15:10:53.798 MQT: stat/tasmota_0B0778/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"1312":"DS18x201"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO24":{"0":"None"},"GPIO25":{"32":"Button1"},"GPIO26":{"224":"Relay1"},"GPIO27":{"0":"None"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}
```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`: none so far
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
14:55:59.152 MQT: stat/tasmota_0B0778/STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_0B0778","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
14:55:59.193 MQT: stat/tasmota_0B0778/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T00:04:12","StartupUTC":"2024-01-24T13:51:47","Sleep":50,"CfgHolder":4617,"BootCount":5,"BCResetTime":"2024-01-24T14:34:31","SaveCount":15}}
14:55:59.236 MQT: stat/tasmota_0B0778/STATUS2 = {"StatusFWR":{"Version":"13.3.0(thermostat_2024.01.01.a)","BuildDateTime":"2024-01-24T14:12:20","Core":"2_0_14","SDK":"4.4.6.231122","CpuFrequency":160,"Hardware":"ESP32-U4WDH-D v3.1","CR":"420/699"}}
14:55:59.266 MQT: stat/tasmota_0B0778/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["saintpierre",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
14:55:59.310 MQT: stat/tasmota_0B0778/STATUS4 = {"StatusMEM":{"ProgramSize":1868,"Free":1011,"Heap":151,"StackLowMark":4,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"164020","FlashFrequency":40,"FlashMode":"DIO","Features":["0809","9F8A97DF","0015A001","B7F7BFCF","05DA9BC4","E0360CC7","480844D2","20200000","D4BC482D","810A80B1","00000000"],"Drivers":"1,2,3,4,5,8,9,10,11,12,14,16,17,20,21,24,29,34,35,38,39,50,52,59,60,62,63,66,67,68,82,86,87,88","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,92,95,98,103,105,109,127","I2CDriver":"7,8,9,10,11,12,13,14,15,17,18,20,24,29,31,36,41,42,44,46,48,58,62,65,69,76,77,82"}}
14:55:59.391 MQT: stat/tasmota_0B0778/STATUS5 = {"StatusNET":{"Hostname":"tasmota-0B0778-1912","IPAddress":"192.168.21.175","Gateway":"192.168.21.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.21.1","DNSServer2":"0.0.0.0","Mac":"C8:2E:18:0B:07:78","IP6Global":"","IP6Local":"fe80::ca2e:18ff:fe0b:778%st1","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"192.168.21.1","DNSServer2":"0.0.0.0","Mac":"00:00:00:00:00:00","IP6Global":"","IP6Local":""},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":16.0}}
14:55:59.443 MQT: stat/tasmota_0B0778/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.21.110","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_0B0778","MqttUser":"YourMqttUser","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
14:55:59.474 MQT: stat/tasmota_0B0778/STATUS7 = {"StatusTIM":{"UTC":"2024-01-24T13:55:59","Local":"2024-01-24T14:55:59","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-28T03:00:00","Timezone":"+01:00","Sunrise":"08:26","Sunset":"17:40"}}
14:55:59.499 MQT: stat/tasmota_0B0778/STATUS10 = {"StatusSNS":{"Time":"2024-01-24T14:55:59","Thermostat1":{"ThermostatModeSet":0,"TempTargetSet":18.00,"CtrDutyCycleRead":0,"ControlMethod":0,"HybridControllerPhase":1,"EmergencyState":0}}}
14:55:59.523 MQT: stat/tasmota_0B0778/STATUS11 = {"StatusSTS":{"Time":"2024-01-24T14:55:59","Uptime":"0T00:04:12","UptimeSec":252,"Heap":153,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":4,"Objects":45},"POWER":"ON","Wifi":{"AP":1,"SSId":"saintpierre","BSSId":"76:A7:41:8D:59:17","Channel":1,"Mode":"11n","RSSI":78,"Signal":-61,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log: 
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by arendst** [comment]
As far as I know we only support the first gen shelly add-on.

I'll order a shelly plus add-on and see if it can work with Tasmota ...

**Comment by Kolia56** [comment]
Thanks for your quick feedback. If you wish I may sponsor the project in order fund this expense.

**Comment by Kolia56** [comment]
FWIW I've a DS18x20 hooked up to the original Shelly fw. It reports temperature as expected

**Comment by Jason2866** [comment]
@Kolia56 Tasmota has a donate button ;-)  Look at the main page of the repo in readme.

**Comment by Kolia56** [comment]
> @Kolia56 Tasmota has a donate button ;-) Look at the main page of the repo in readme.

I've been searching it for a while. Thanks. Button not big enough ;). Done

**Reply by arendst** [reply]
Thx. Got it.

**Comment by arendst** [comment]
Progress!!

```
00:00:00.003 HDW: ESP32-U4WDH-S v3.0 
00:00:00.010 UFS: FlashFS mounted with 304 kB free
00:00:00.016 CFG: Loaded from File, Count 139
00:00:00.022 QPC: Count 1
00:00:00.028 ROT: Mode 1
00:00:00.058 CFG: No '*.autoconf' file found
00:00:00.062 BRY: Berry initialized, RAM used 3711 bytes
00:00:00.069 BRY: No 'preinit.be'
00:00:00.077 CFG: Energy loaded from file
00:00:00.077 NRG: Init driver 1
00:00:00.078 SRC: Restart
00:00:00.083 Project tasmota - ShellyPlus1PM Version 13.3.0.4(theo)-2_0_14(2024-01-25T16:12:07)
00:00:00.083 ETH: No ETH MDC and/or ETH MDIO GPIO defined
00:00:00.100 DSB: Sensors found Test 1
00:00:00.114 BRY: No 'autoexec.be'
00:00:00.253 WIF: Attempting connection...
00:00:01.004 WIF: Connecting to AP1 indebuurt_IoT Channel 11 BSSId 18:E8:29:CA:17:C1 in mode 11n as shellyplus1pm...
00:00:01.070 WIF: IPv4 192.168.2.182, mask 255.255.255.0, gateway 192.168.2.254
00:00:02.980 WIF: Connected
00:00:03.432 WIF: IPv6 Local fe80::aa03:2aff:feb7:7218%st1
00:00:04.006 NTP: No reply from 127.0.1.1
00:00:04.040 HTP: Web server active on shellyplus1pm with IP address 192.168.2.182
00:00:04.076 RTC: UTC 2024-01-25T15:13:31Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
16:13:31.001 RTC: Synced by NTP
16:13:32.069 MQT: Attempting connection...
16:13:32.097 MQT: Connected
16:13:32.103 MQT: tele/shellyplus1pm/LWT = Online (retained)
16:13:32.105 MQT: cmnd/shellyplus1pm/POWER = 
16:13:32.106 MQT: Subscribe to cmnd/shellyplus1pm/#
16:13:32.107 MQT: Subscribe to cmnd/tasmotas/#
16:13:32.108 MQT: Subscribe to cmnd/DVES_B77218_fb/#
16:13:32.111 MQT: tele/shellyplus1pm/INFO1 = {"Info1":{"Module":"Shelly Plus 1PM Add-On","Version":"13.3.0.4(theo)single-core","FallbackTopic":"cmnd/DVES_B77218_fb/","GroupTopic":"cmnd/tasmotas/"}}
16:13:32.116 MQT: tele/shellyplus1pm/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"shellyplus1pm","IPAddress":"192.168.2.182","IP6Global":"","IP6Local":"fe80::aa03:2aff:feb7:7218%st1"}}
16:13:32.119 MQT: tele/shellyplus1pm/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":49}}
16:13:32.122 MQT: stat/shellyplus1pm/RESULT = {"POWER":"ON"}
16:13:32.123 MQT: stat/shellyplus1pm/POWER = ON
16:13:32.126 TFS: File 'autoexec.bat' not found
16:13:34.008 QPC: Reset
16:13:34.432 WIF: IPv6 Global fd84:7996:2cbf:1349:aa03:2aff:feb7:7218
16:13:36.014 APP: Boot Count 49
16:13:36.741 CFG: Saved, Count 140, Bytes 4096
16:13:37.009 MQT: tele/shellyplus1pm/STATE = {"Time":"2024-01-25T16:13:36","Uptime":"0T00:00:10","UptimeSec":10,"Heap":180,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":32,"MqttCount":1,"Berry":{"HeapUsed":3,"Objects":34},"POWER":"ON","Wifi":{"AP":1,"SSId":"indebuurt_IoT","BSSId":"18:E8:29:CA:17:C1","Channel":11,"Mode":"11n","RSSI":100,"Signal":-36,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:13:37.030 MQT: tele/shellyplus1pm/SENSOR = {"Time":"2024-01-25T16:13:37","Switch1":"OFF","ENERGY":{"TotalStartTime":"2022-10-04T12:24:14","Total":0.000,"Yesterday":0.000,"Today":0.000,"Period":0.00,"Power":0.00,"ApparentPower":0.00,"ReactivePower":0.00,"Factor":0.00,"Voltage":226.4,"Current":0.000},"DS18B20":{"Id":"0516707FA6FF","Temperature":22.2},"TempUnit":"C"}
```

![image](https://github.com/arendst/Tasmota/assets/11044339/e1c660f7-a784-4d27-8d4c-4981e87cab9b)

**Comment by Kolia56** [comment]
What an achievement in a so short time. Kudos. Shall I try it or still wait? If yes what should I download? Thank you.

**Comment by wjohn007** [comment]
Great news on this topic.

**Comment by kroon040** [comment]


**Reply by Kolia56** [reply]
This discussion is for general support of Shelly plus add-on, not specific sensors. Please open a new discussion if you wish and delete your post. Thank you.

**Comment by arendst** [comment]
Use latest dev branch with support for DS18x20 on Shelly Plus Add-On.

Using this template ```{"NAME":"Shelly Plus 1PM Add-On","GPIO":[1,1,0,1,192,2720,0,0,0,0,0,0,0,0,2656,1,0,0,0,2624,0,0,224,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}``` one can configure for DS18x20 using Configure Module:

![image](https://github.com/arendst/Tasmota/assets/11044339/3b84a430-8949-489f-8e9a-34651f639880)

Results in 

![image](https://github.com/arendst/Tasmota/assets/11044339/037e6185-613c-4a9c-b6a0-6d086e6b9491)

**Reply by HB787** [reply]
Good news, would like the template for the  Shelly 2PM PCB v0.1.9 in combination with the add-on.

**Reply by arendst** [reply]
Just go to Configure Template and change the following GPIOs from ``None`` to ``User``:

GPIO0, GPIO1, GPIO3 and GPIO19.

Save it and let it restart. Then using Configure Module you can select a supported device on supported GPIOs. You'll need to know that GPIO0 is always an output channel so select either ``DS18x20_o1`` or ``DHT11_o1``. Then on GPIO1 select either ``DS18x201`` or one of ``DHT11``, ``AM2301`` or ``SI7021`` devices.

GPIO19 could be configured as ``button1``.

GPIO3 is an "analog" input channel but currently it's not supported by Tasmota.

**Reply by HB787** [reply]
Thanks, it worked

**Reply by HB787** [reply]
It worked for a couple of hours but seems to lose the connection somehow.
I have a DHT11 device connected and tried these settings.

**Reply by Noschvie** [reply]
My "Shelly Plus 1 Add-On" and a DS18x20 connected is running for over 20 hours...

**Reply by Kolia56** [reply]
Same with me, has been working steadily for at least 2 days

**Reply by arendst** [reply]
The issue is not so much the Add-On but the crappy timing sensitive DHT11.

To ease the pain dive into the ``DhtDelay`` command and find an optimum for your device.

**Reply by pgorod** [reply]
This probably doesn't apply anymore with this new firmware, but with the old, incompatible Tasmota, I used to get overheating when I connected the add-on, then eventually a burnt device. It could take days to burn.

I'm just saying this now, because if anyone has unstable behaviours, it would be a good idea to check temperatures. Simply touch it, or read the internal temperature sensors (which are _not_ the same as the DS1820 you might be trying to connect)

**Comment by Kolia56** [comment]
Thank you. DS18x20_o is not listed in the menu. I compiled from the current development branch with env set to tasmota32. My unit is a dual core. Could it make a difference?

![image](https://github.com/arendst/Tasmota/assets/20872098/8684754e-75de-438f-a115-c29c426e68a0)

```
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP32
Chip is ESP32-U4WDH (revision v3.1)
Features: WiFi, BT, Dual Core, 240MHz, Embedded Flash, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: c8:2e:18:0b:07:78
Stub is already running. No upload is necessary.
Manufacturer: 20
Device: 4016
Detected flash size: 4MB
Hard resetting via RTS pin...
```

**Reply by arendst** [reply]
> DS18x20_o is not listed in the menu

In that case you're not using the latest changes. Pls recheck https://github.com/arendst/Tasmota/commit/4bc7b4ec2f8e77e65fd8bb115a181cb56cd5d383

**Comment by Kolia56** [comment]
> > DS18x20_o is not listed in the menu
> 
> In that case you're not using the latest changes. Pls recheck [4bc7b4e](https://github.com/arendst/Tasmota/commit/4bc7b4ec2f8e77e65fd8bb115a181cb56cd5d383)

You are right. I have now two working DS18x20 sensors. Thank you again. I will start to set-up thermostat module, this is the whole idea behind having temp sensors hooked-up to this module.  As soon the new Tasmota release is on air, I will update Github template site. May I suggest to do the same with the TEMPLATES.md file?

**Reply by arendst** [reply]
> May I suggest to do the same with the TEMPLATES.md file?

This file is automatically build (by me) from the template site just before a new release.

**Comment by Kolia56** [comment]
Shall I do it for the template site now then?

**Reply by Jason2866** [reply]
I know no reason why not. Just mention that currently the development version needs to be used.

**Comment by Kolia56** [comment]
As a side comment, I confirm that Thermostat current temperature sensor is updated by the local sensor. Just make sure to use the local temperature sensor.
```
sensorinputset 1
```
The default name for the sensor is DS18B20 which is fine as long as there is only one sensor connected to the add-on. In case several sensors are connected, there will be named DS18B20-1, DS18B20-2, etc. In that case, the code has to be changed accordingly before compilation.

**Comment by Noschvie** [comment]
Hello,
thanks for these modifications!

How have the template for the "Shelly plus 1" be changed to get the DS18B20 working?
Thanks!

**Reply by sfromis** [reply]
You can see the two pins needed here: https://github.com/arendst/Tasmota/discussions/20580#discussioncomment-8256030

**Reply by Noschvie** [reply]
> You can see the two pins needed here: https://github.com/arendst/Tasmota/discussions/20580#discussioncomment-8256030

Ok, got it up and running, thanks again.

**Comment by Kolia56** [comment]
@arendst Thanks again for your dedication to this incredible software and its related documentation.

**Comment by kroon040** [comment]
Can is ask in this discussion if gpio1 and gpio19 can be used for the fingerprint sensor (r503/as608) for rx and tx? 

Also the motion sensor HC-SR501, can this als be used on the addon?

**Reply by Jason2866** [reply]
Yes, asking is possible. Just do not hijack a discussion where "other" stuff is discussed. So please open a new one.

**Reply by kroon040** [reply]
Ok I open a new discussion

**Comment by 2aannddrree2** [comment]
Hello i can´t connect a ADC on GPIO3,
after rebot of the device disappears the ADC config ?

![image](https://github.com/user-attachments/assets/07d19087-f533-4d80-a9fc-f58d8bb861b2)

**Reply by sfromis** [reply]
Don't expect that the add-on is doing nothing to the signal.

Specifically, it will not pass low voltage analog, as the pins directly on the Shelly Plus 1PM will have mains AC voltage levels, like about 236 V on the "3.3V" pin and 233 V on the "GND" pin (or whatever you mains happens to be at any moment in time). While the add-on protects you from that, you do not get analog pass-thru.

Configs going away is usually because what you did was bad enough to cause a boot loop, where Tasmota can unconfigure features until able to boot without quickly crashing yet again.

**Reply by 2aannddrree2** [reply]
How can i connect an analog input on the Shelly add on with Tasmota? 

In this post https://github.com/arendst/Tasmota/discussions/18885#discussioncomment-7295482 i can read ==> ANALOG IN = GPIO3

**Reply by sfromis** [reply]
Not tried it, but *as far as I've understood*, you cannot get an analog signal passed to the ESP chip. Instead the add-on has a circuit to generate a sawtooth signal with the duty cycle matching the analog value. https://github.com/arendst/Tasmota/discussions/21248

**Reply by 2aannddrree2** [reply]
Strange with the original shelly firmware it works very well

**Reply by sfromis** [reply]
Sure, with the feature of measuring duty cycle. It cannot be using analog input on the ESP32. And as discussed in the linked thread, timing a duty cycle is not a current feature of Tasmota.

**Comment by gmcpaul** [comment]
Hi, 
i have a Shelly Plus 1 with the Addon Attached.
What config should i use, so i can measure ds18b20's ?

that one?
https://templates.blakadder.com/shelly_plus_add-on.html
>> with that config im having the problem that the gpio's are not appearing in the config section


or just apply https://github.com/arendst/Tasmota/discussions/20580#discussioncomment-8256030 and let it pretend it is a shelly plus 1 pm`?
>> this is showing the gpios but i have no data from it

sorry... but im a little confused

**Reply by gmcpaul** [reply]
@arendst  sorry to disturb but maybe you can shade some light to that

**Reply by arendst** [reply]
Using this template on a Shelly Plus 1PM
```
{"NAME":"Shelly Plus 1PM Add-On","GPIO":[1,1,0,1,192,2720,0,0,0,0,0,0,0,0,2656,1,0,0,0,2624,0,0,224,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}
```
it works as expected

![image](https://github.com/user-attachments/assets/0bae4fb7-0f92-4eb5-8428-50bd8cce1fe7)

![image](https://github.com/user-attachments/assets/e3d740bf-63a3-4125-8c50-40d4af46302c)

So I guess using this template on a Shelly Plus 1
```
{"NAME":"Shelly Plus 1","GPIO":[1,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,32,224,0,0,0,0,0,4736,4705,0,0,0,0,0,0],"FLAG":0,"BASE":1}
```
would work the same...

**Reply by gmcpaul** [reply]
Hi Theo,
i noticed that i was on an older Firmware Version.
after i managed to update to the latest release, all went well.

thanks again and best regards
paul

**Comment by aizerin** [comment]
Does this add-on works also for Shelly 1PM Gen3? According to Shelly docs it should be compatible. But 1PM has different pins than plus.

**Reply by sfromis** [reply]
Shelly has a page detailing supported devices:
https://kb.shelly.cloud/knowledge-base/shelly-plus-add-on

**Comment by madduck** [comment]
I've been trying to get the Shelly Plus 1PM & Add-On to work with DS18B20 (with a 4.7kΩ resistor between VCC and Data), using Tasmota 15.x, but no luck. I've tried GPIOs 0/1, 2/3 and 19/20, I've tried with and without `ds18x20_o`, but it just won't see the sensor ("DSB: Sensors found 0").

I have it working with a Wroom32, connecting data directly to GPIO23, so the sensor is working. I've also tried two differen PM1 and two different add-ons, to rule out problems there, but it's the same with any combination.

Does anyone have a working template for this, as the ones listed above don't work.
